code/modules/atmospherics/machinery/components/components_base.dm

```diff
@@ -134,7 +134,7 @@
 		air_update_turf(1)
 
 /obj/machinery/atmospherics/components/proc/safe_input(var/title, var/text, var/default_set)
-	var/new_value = input(usr,text,title,default_set) as num
+	var/new_value = input(usr,text,title,default_set) as num|null
```

Comment:
should probably return `default_set` when cancelled

Reply:
`|| default_set` instead of sep check